legend-sdlc-server/src/main/java/org/finos/legend/sdlc/server/gitlab/api/GitLabReviewApi.java

```diff
@@ -139,6 +140,13 @@ else if (revisionIds instanceof Set)
             {
                 // if no revision ID is specified we will use the default merge request API from Gitlab to take advantage of the filter
                 MergeRequestFilter mergeRequestFilter = withMergeRequestFilters(new MergeRequestFilter(), state, since, until).withProjectId(gitLabProjectId.getGitLabId());
+                if (sources != null)
+                {
+                    for (WorkspaceSource source : sources)
+                    {
+                       mergeRequestFilter.withTargetBranch(getSourceBranch(gitLabProjectId, source));
+                    }
```

Comment:
It's only valid to set the target branch on the MergeRequestFilter if there's only one source. Also, if there are any number of sources, there needs to be a filter on the stream of MRs that comes back.